src/OrchardCore.Modules/OrchardCore.Resources/Liquid/ResourcesTag.cs

```diff
@@ -16,69 +16,30 @@ public class ResourcesTag
         public static async ValueTask<Completion> WriteToAsync(List<FilterArgument> argumentsList, TextWriter writer, TextEncoder _, TemplateContext context)
         {
             var services = ((LiquidTemplateContext)context).Services;
-            var resourceManager = services.GetRequiredService<IResourceManager>();
+            var processors = services.GetRequiredService<IEnumerable<IResourcesTagHelperProcessor>>();
 
-            var type = ResourceType.Footer;
+            var processorContext = new ResourcesTagHelperProcessorContext(writer, ResourceTagType.Footer);
 
             foreach (var argument in argumentsList)
             {
                 switch (argument.Name)
                 {
-#pragma warning disable CA1806 // Do not ignore method results
-                    case "type": Enum.TryParse((await argument.Expression.EvaluateAsync(context)).ToStringValue(), out type); break;
-#pragma warning restore CA1806 // Do not ignore method results
+                    case "type":
+                        var typeString = (await argument.Expression.EvaluateAsync(context)).ToStringValue();
+                        if (Enum.TryParse<ResourceTagType>(typeString, out var type))
+                        {
+                            processorContext = processorContext with { Type = type };
+                        }
+                        break;
                 }
             }
 
-            switch (type)
+            foreach (var processor in processors)
```

Comment:
Why the foreach here, while it's done inside the tag helper

Reply:
Sorry, I don't understand the question. This is the Liquid tag, it's independent of `ResourcesTagHelper` which is for Razor.